lib/load-option.js

```diff
@@ -12,4 +12,14 @@ module.exports = function loadOption (name) {
   if (pkg === 'true') return true
   if (pkg === 'false') return false
   if (pkg) return pkg
+
+  if (process.env['npm_package_scripty_' + name + '_0']) {
```

Comment:
hah, TIL how npm handles array types in its env var unrolling of nested package.json props.

This method was already a little crusty before this addition, but its intent is definitely too mysterious now. I'd recommend yanking out a few private functions at least that reveal intent first, and then adding this.


If this file had previously looked like the following listing, how might you name & factor this new behavior?

```js
var _ = require('lodash')

module.exports = function loadOption (name) {
  if (envVarSet(posixEnvVarName(name))) {
    return boolEnvVarValue(posixEnvVarName(name))
  } else if (envVarSet(packageEnvVarName(name))) {
    return boolEnvVarValue(packageEnvVarName(name))
  }
}

function boolEnvVarValue (envVarName) {
  var value = process.env[envVarName]

  if (value === 'true') {
    return true
  } else if (value === 'false') {
    return false
  } else {
    return value
  }
}

function envVarSet (envVarName) {
  return !!process.env[envVarName]
}

function posixEnvVarName (optionName) {
  return 'SCRIPTY_' + _.snakeCase(optionName).toUpperCase()
}

function packageEnvVarName (optionName) {
  return 'npm_package_scripty_' + optionName
}
```

Reply:
I have updated my PR with this change plus my new ones on top of that